cli/cmd/uninstall.go

```diff
@@ -0,0 +1,288 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/ghodss/yaml"
+	"github.com/linkerd/linkerd2/pkg/k8s"
+	"github.com/spf13/cobra"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	apiregistrationv1client "k8s.io/kube-aggregator/pkg/client/clientset_generated/clientset/typed/apiregistration/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	yamlSep                   = "---\n"
+	rbacVersion1              = "rbac.authorization.k8s.io/v1"
+	apiExtensionVersion1Beta1 = "apiextensions.k8s.io/v1beta1"
+	coreVersion1              = "v1"
+	extensionsVersion1Beta1   = "extensions/v1beta1"
+	admissionRegVersion1Beta1 = "admissionregistration.k8s.io/v1beta1"
+	apiRegVersion1            = "apiregistration.k8s.io/v1"
+)
+
+type kubernetesResource struct {
+	runtime.TypeMeta
+	metav1.ObjectMeta `json:"metadata"`
+}
+
+func newKubernetesResource(apiVersion, kind, name string) kubernetesResource {
+	r := kubernetesResource{}
```

Comment:
I think we can inline this struct initialization:

```
return kubernetesResource {
  runtime.TypeMeta {
    APIVersion: apiVersion,
    Kind: kind,
  },
  metav1.ObjectMeta {
    Name: name,
  },
}
```

Reply:
Changed. Good to know how embedded types behave in go, thought (for some reason) that wouldn't work quite well, but turns out it does 👍 .